### PR TITLE
Extract error modal into a separate component

### PR DESCRIPTION
**Pull request by rohithbalaji123**
### Description
The error modal is written inside HomePage and it maintains all the state related to any errors existing. There is a need to be able to reuse the component as there are more pages(Admin panel) getting added into the project.

### Description
The error modal has been separated into a component which takes care of listening to the global `err` state and pop up a modal when it has been raised. The page requiring this functionality just needs to add the component without worrying about it's implementation. It promptly subscribes to the redux state changes when mounted and unsubscribes when unmounted. And it is very easy to reuse.

### CODE REVIEW

File: frontend/src/component/styles/color.scss [hunk 1]
```diff
@@ -24,6 +24,9 @@ $color-page-indicator-active-hover-text: white;
 
 $color-admin-panel-background: #f6f6f6;
 
+// Error Modal
+$color-error-modal-background-color: #424242;
```
Comment (rohithbalaji123) [inline frontend/src/component/styles/color.scss:28]:
```suggestion
$color-error-modal-background: #424242;
```

File: frontend/src/component/pages/shared/ErrorModal.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  constructor(props: IProps) {
+    super(props);
+
+    this.state = {
+      error: props.store.getState().err
+    };
+  }
+
+  componentDidMount(): void {
+    this.startErrorListener();
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/ErrorModal.tsx:32]:
```suggestion
    this.subscribeErrorChanges();
```

File: frontend/src/component/pages/shared/ErrorModal.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  private startErrorListener() {
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/ErrorModal.tsx:35]:
```suggestion
  private subscribeErrorChange() {
```

File: frontend/src/component/pages/shared/ErrorModal.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    this.stopErrorListener = this.props.store.subscribe(() => {
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/ErrorModal.tsx:36]:
```suggestion
    this.unsubscribeStateChange = this.props.store.subscribe(() => {
```

Comment (byliuyang) [reply frontend/src/component/pages/shared/ErrorModal.tsx:36]: This is just for consistency with the the type definition.

File: frontend/src/component/pages/shared/ErrorModal.tsx [hunk 5]
```diff
@@ -29,11 +29,11 @@ export class ErrorModal extends Component<IProps, IState> {
   }
 
   componentDidMount(): void {
-    this.startErrorListener();
+    this.subscribeErrorChanges();
   }
 
-  private startErrorListener() {
-    this.stopErrorListener = this.props.store.subscribe(() => {
+  private subscribeErrorChanges() {
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/ErrorModal.tsx:35]:
```suggestion
  private subscribeErrorChange() {
```

File: frontend/src/component/pages/shared/ErrorModal.tsx [hunk 6]
```diff
@@ -29,11 +29,11 @@ export class ErrorModal extends Component<IProps, IState> {
   }
 
   componentDidMount(): void {
-    this.startErrorListener();
+    this.subscribeErrorChanges();
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/ErrorModal.tsx:32]:
```suggestion
    this.subscribeErrorChange();
```

**Pull request merged**